ui/src/App.tsx

```diff
@@ -235,28 +195,60 @@ const App = () => {
         </div>
       </Box>
       <ChakraProvider>
-        <Modal isOpen={showOverlay} onClose={() => setShowOverlay(false)}>
+        <Modal isOpen={showOverlay} onClose={closeOverlay}>
           <ModalOverlay />
           <ModalContent>
             <ModalHeader>Event Details</ModalHeader>
             <ModalCloseButton />
-            <ModalBody>
-              <p>Event title: {displayedEventData.title}</p>
-              <p>Description: {displayedEventData.description}</p>
-              <p>
-                Start:{" "}
-                {modalDateFormat(new Date(displayedEventData.startTimeUtc))}
-              </p>
-              <p>
-                End: {modalDateFormat(new Date(displayedEventData.endTimeUtc))}
-              </p>
-            </ModalBody>
-
-            <ModalFooter>
-              <Button onClick={handleDeleteEntry} variant="ghost">
-                Delete
-              </Button>
-            </ModalFooter>
+            {!inEditMode && (
+              <>
+                <ModalBody>
+                  <p>Event title: {displayedEventData.title}</p>
+                  <p>Description: {displayedEventData.description}</p>
+                  <p>
+                    Start:{" "}
+                    {modalDateFormat(new Date(displayedEventData.startTimeUtc))}
+                  </p>
+                  <p>
+                    End:{" "}
+                    {modalDateFormat(new Date(displayedEventData.endTimeUtc))}
+                  </p>
+                </ModalBody>
+                <ModalFooter>
+                  <Button onClick={handleEditEntry} variant="ghost">
+                    Edit
+                  </Button>
+                  <Button onClick={handleDeleteEntry} variant="ghost">
+                    Delete
+                  </Button>
+                </ModalFooter>
+              </>
+            )}
+            {inEditMode && (
+              <ModalBody>
+                <EventForm
+                  initialTitle={displayedEventData.title}
+                  initialStartDate={formatDate(
+                    new Date(displayedEventData.startTimeUtc)
+                  )}
+                  initialEndDate={formatDate(
+                    new Date(displayedEventData.endTimeUtc)
+                  )}
+                  initialStartTime={`${padNumberWith0Zero(
```

Comment:
What do you think of making helper functions for these repeated formattings? something like `formatTime`